client/components/withModal.jsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import Modal from './Modal.jsx';
+
+const withModal = WrappedComponent => {
+    class _WithModal extends React.Component {
+        state = {
+            isOpen: false
+        };
+        onShow = e => {
+            this.setState({
+                isOpen: true
+            });
+        };
+        onClose = e => {
+            this.setState({
+                isOpen: false
+            });
+        };
+        render() {
+            return (
+                <div
+                    onClick={e => {
+                        this.onShow(e);
+                    }}
+                >
+                    <WrappedComponent {...this.props} />
```

Comment:
I'm also curious about why there's one Wrapped component here, and another below this line nested inside of a <Modal/> component.